src/neoxp/Node/NodeUtility.cs

```diff
@@ -138,5 +142,127 @@ public static void SignOracleResponseTransaction(ProtocolSettings settings, Expr
+            }
+            
+            using var rpcClient = new RpcClient(uri);
+            var stateAPI = new StateAPI(rpcClient);
+                
+            uint height = stateHeight;
+            if (height == 0)
+            {
+                var stateHeight_ = await stateAPI.GetStateHeightAsync();
+                if (stateHeight_.localRootIndex is null)
+                {
+                    throw new Exception("Null \"localRootIndex\" in state height response");
+                }
+                height = stateHeight_.localRootIndex.Value;
+            }
+
+            var stateRoot = await stateAPI.GetStateRootAsync(height);
+            var states = await rpcClient.ExpressFindStatesAsync(stateRoot.RootHash, contractHash_, new byte[0]);
+            var contractState = await rpcClient.GetContractStateAsync(contractHash).ConfigureAwait(false);
+
+            return (contractState, states.Results);
+        }
+            
+        
+        public static int PersistContract(SnapshotCache snapshot, ContractState state, (string key, string value)[] storagePairs, ContractCommand.ContractForce? force)
+        {
+            var localContract = NativeContract.ContractManagement.GetContract(snapshot, state.Hash);
+            
+            if (localContract is null)
+            {
+                // Our local chain might already be using the contract id of the pulled contract, we need to check for this
+                // to avoid having contracts with duplicate id's. This is important because the contract id is part of the
+                // StorageContext used with Storage syscalls and else we'll potentially override storage keys or iterate
+                // over keys that shouldn't exist for one of the contracts.
+                if (state.Id < LastUsedContractId(snapshot))
+                {
+                    state.Id = GetNextAvailableId(snapshot);
+                }
+                else
+                {
+                    // Update available id such that a regular contract deploy will use the right next id;
+                    SetLastUsedContractId(snapshot, state.Id);
+                }
+
+                StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_Contract).Add(state.Hash);
+                snapshot.Add(key, new StorageItem(state));
+            }
+            else 
+            {
+                if (force is null)
+                {
+                    throw new Exception("Contract already exists locally. Use --force:<option> to overwrite");
+                }
+                if (force == ContractCommand.ContractForce.All ||
+                    force == ContractCommand.ContractForce.ContractOnly)
+                {
+                    // Note: a ManagementContract.Update() will not change the contract hash. Not even if the NEF changed.
+                    localContract.Nef = state.Nef;
```

Comment:
I don't think setting localContract fields automatically saves the data back the storage. You're only modifying the in-memory copy you got from calling `GetContract`

Reply:
oof that was sloppy from my side

Reply:
Have we validated that setting the value returned from GetInteroperable also updates the raw value? I don't think it does

Reply:
I have validated this